katdal/test/test_chunkstore_s3.py

```diff
@@ -120,7 +120,14 @@ def array_name(self, path):
         bucket = 'katdal-unittest'
         return self.store.join(bucket, path)
 
+    @timed(0.1 + 0.1 + 1 + 0.05)
     def test_store_unavailable(self):
         # Drastically reduce the default botocore timeout of nearly 7 seconds
         assert_raises(StoreUnavailable, S3ChunkStore.from_url,
-                      'http://apparently.invalid/', timeout=0.1)
+                      'http://apparently.invalid/',
+                      timeout=0.1, extra_timeout=0)
+        # Some pathological DNS setups (sshuttle?) ignore the original timeout,
+        # preferring to wait 25-30 seconds instead... Check that it is fixed.
+        assert_raises(StoreUnavailable, S3ChunkStore.from_url,
+                      'http://a-valid-domain-is-somehow-harder.kat.ac.za/',
+                      timeout=0.1, extra_timeout=1)
```

Comment:
I don't see how this is testing a slow `getaddrinfo` - surely you need to mock it?

Reply:
How does one mock? :-P Never mind, I'll figure it out.

I've been testing it by running it on my pathological setup at home but mocking is the sensible way. You'll make an avid tester out of me yet!

Reply:
Mocked.